### PR TITLE
Add `trie-compact` command

### PR DESCRIPTION
**Pull request by georgepisaltu**
This PR adds a new `compact-trie` subcommand to `casper-db-utils` and builds on top of https://github.com/casper-network/casper-node/pull/2948.

In addition to that functionality, this PR:
- moves `trie-compact` code to a standalone module
- makes errors more explicit
- revamps the CLI API
- adds checks throughout the code and propagates errors
- adds tests for `compact-trie`

PR was opened with a clippy error, I will push a commit fixing it along with other potential fixes after reviews are done.

Added @sacherjj as a reviewer just to chime in on the CLI API of the tool.

### CODE REVIEW

**Review by dwerner — APPROVED** [review]
Looking really good to me! Thanks for taking this on and improving it. :+1:

File: src/main.rs [hunk 1]
```diff
@@ -62,6 +76,63 @@ fn main() {
...
                 )
         )
+        .subcommand(
+            Command::new(COMPACT_TRIE)
+                .about("Writes a compacted version of the block entries in the source trie store to the destination.")
+                .arg(
+                    Arg::new(APPEND)
+                        .required(false)
+                        .short('a')
+                        .long(OVERWRITE)
```
Comment (Fraser999) [inline src/main.rs:86]:
```suggestion
                        .long(APPEND)
```

File: src/main.rs [hunk 2]
```diff
@@ -62,6 +76,63 @@ fn main() {
...
+                        .help("Path of the directory where the output `data.lmdb` file will be created."),
+                )
+                .arg(
+                    Arg::new(OVERWRITE)
+                        .required(false)
+                        .short('w')
+                        .long(OVERWRITE)
+                        .takes_value(false)
+                        .conflicts_with(APPEND)
+                        .help("Append output to an already existing output `data.lmdb` file in destination directory."),
```
Comment (Fraser999) [inline src/main.rs:106]: This looks like a copy-paste error.

File: src/main.rs (hunk 1 above)
Comment (georgepisaltu) [reply src/main.rs]: Done.

File: src/main.rs (hunk 2 above)
Comment (georgepisaltu) [reply src/main.rs]: Fixed it.

**Review by Fraser999 — APPROVED** [review]
LGTM - happy to approve now, but we'll need to test on mainnet and testnet data before publishing it.

**Comment by sacherjj** [comment]
New logging is nice.  Looks like it is working other than the startup db move.  Will follow up when complete.

**Pull request merged**